internal/hud/server/server.go

```diff
@@ -80,3 +86,16 @@ func (s HeadsUpServer) HandleAnalytics(w http.ResponseWriter, req *http.Request)
 		s.a.Incr(p.Name, p.Tags)
 	}
 }
+
+func (s HeadsUpServer) HandleSail(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		http.Error(w, "must be POST request", http.StatusBadRequest)
+		return
+	}
+
+	err := s.sailCli.Connect(req.Context(), s.store)
+	if err != nil {
+		s.store.Dispatch(store.NewErrorAction(errors.Wrap(err, "sailClient")))
```

Comment:
this will make Tilt crash, which is probably not what you want. would be better to return an http.Error (like on the line above) and have the UI handle the error properly.

Reply:
😅